### ISSUE TITLE
Target sanity.external(extended.external) with hotspot and no parallel build get the build error with external_custom project

### ISSUE THREAD

**Issue by sophia-guo** [adoptium/aqa-tests#3319]
Target sanity.external with hotspot and no parallel build get the build error with external_custom project, which is found by grinder:
https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3454/console

```
08:54:29  build_image:
08:54:29       [echo] Executing external.sh --build --dir external_custom --tag nightly --version 8 --impl hotspot 
08:54:29       [exec] The test here is external_custom
08:54:29       [exec] The directory in the external.sh is 
08:54:29       [exec] The supported tests are external_custom camel derby elasticsearch jacoco jenkins functional-test kafka lucene-solr openliberty-mp-tck payara-mp-tck quarkus quarkus_quickstarts scala system-test tomcat tomee wildfly wycheproof netty spring
08:54:29       [exec] 
08:54:29       [exec] usage: /home/jenkins/workspace/Grinder/aqa-tests/TKG/../../jvmtest/external/external.sh test version vm os package build check_external_custom [testtarget]
08:54:29       [exec] test    = external_custom camel derby elasticsearch jacoco jenkins functional-test kafka lucene-solr openliberty-mp-tck payara-mp-tck quarkus quarkus_quickstarts scala system-test tomcat tomee wildfly wycheproof netty spring
08:54:29       [exec] version = 8 11 17
08:54:29       [exec] vm      = hotspot openj9
08:54:29       [exec] os      = ubuntu
08:54:29       [exec] package = jdk jre
08:54:29       [exec] build   = full
08:54:29       [exec] testtarget = Optional: CMD to pass to ENTRYPOINT script from Dockerfile
08:54:29  
08:54:29  BUILD FAILED
08:54:29  /home/jenkins/workspace/Grinder/aqa-tests/TKG/scripts/build_test.xml:95: The following error occurred while executing this line:
08:54:29  /home/jenkins/workspace/Grinder/aqa-tests/external/build.xml:42: The following error occurred while executing this line:
08:54:29  /home/jenkins/workspace/Grinder/aqa-tests/external/external_custom/build.xml:42: The following error occurred while executing this line:
08:54:29  /home/jenkins/workspace/Grinder/aqa-tests/external/build.xml:87: exec returned: 255
```

This only happened to hotspot as build process is disable for openj9 and ibm and only happened when target is sanity.external or extended.external without setting EXTERNAL_CUSTOM_REPO

**Comment by smlambert** [comment]
@LizyBbethy - if you can take a look at this next, it will be great!

**Comment by LizyBbethy** [comment]
i am up for it

**Comment by LizyBbethy** [comment]
@smlambert @sophia-guo  does this call for setting some default values for the External_custom app  to enable it work ?

**Comment by smlambert** [comment]
Yes that is definitely one way to approach it (and it can be the first way we approach it).  

@LizyBbethy - can you add some code in the [external_custom/build.xml](https://github.com/adoptium/aqa-tests/blob/master/external/external_custom/build.xml#L12-L18) file which checks, if EXTERNAL_CUSTOM_REPO is not set, then set EXTERNAL_CUSTOM_REPO=https://github.com/adoptium/bumblebench and then run a Grinder.

From that you will see that the ant script (build.xml) will likely pass.  Then you can consider if there are other ways to handle the case where EXTERNAL_CUSTOM_REPO is not set.

**Comment by sophia-guo** [comment]
if EXTERNAL_CUSTOM_REPO is set with default value build process should be fine.
However with the default set both build process and test running process are not necessary ( as not application we are interested in are tested). As external_custom is designed to be used by user specific( won't be necessary a part of regular run) skipping it might be a good solution. 

To skip it we need skip it in build and run stage
1. build process: in the [build target](https://github.com/adoptium/aqa-tests/blob/master/external/external_custom/build.xml#L12-L18) if EXTERNAL_CUSTOM_REPO is not set do nothing ( or exit ).
2. running test: update level to [special](https://github.com/adoptium/aqa-tests/blob/master/external/external_custom/playlist.xml#L36])). Could also just delete MachineInfo testcase, which avoids to fail the build if there is no testcase for the whole build. If external_custom is only used for user specific there is no reason to keep MachineInfo as a testcase holder. (In this case with the dynamic compile  probably we don't even need step one, however with step one it would be clearer and rigorous.)


@renfeiw For dynamic compile, if under external/external_custom there is only one testcase with level special. If the target is sanity.external or extended.external if external/external_custom will be compiled?

**Comment by smlambert** [comment]
I think we should raise a separate issue for the removal of MachineInfo targets.

**Comment by sophia-guo** [comment]
Issue raised. https://github.com/adoptium/aqa-tests/issues/3334

**Comment by renfeiw** [comment]
>@renfeiw For dynamic compile, if under external/external_custom there is only one testcase with level special. If the target is sanity.external or extended.external if external/external_custom will be compiled?

In this case, the external/external_custom will NOT be compiled.

**Comment by LizyBbethy** [comment]
> > @renfeiw For dynamic compile, if under external/external_custom there is only one testcase with level special. If the target is sanity.external or extended.external if external/external_custom will be compiled?
> 
> In this case, the external/external_custom will NOT be compiled.

That implies the approach might not just be it

**Comment by smlambert** [comment]
https://github.com/adoptium/aqa-tests/issues/3319 and this are reporting the same issue.  I will close 1 of them as duplicate.

**Comment by sophia-guo** [comment]
@LizyBbethy  see if the  change helps you understand the issue
https://github.com/adoptium/aqa-tests/compare/master...sophia-guo:health?expand=1

**Comment by sophia-guo** [comment]
PR created #3383 as this blocks #3187

**Comment by sophia-guo** [comment]
Closed by #3383
